astropy/units/quantity_helper/function_helpers.py

```diff
@@ -904,14 +912,35 @@ def interp(x, xp, fp, *args, **kwargs):
 
 @function_helper
 def unique(
-    ar, return_index=False, return_inverse=False, return_counts=False, axis=None
+    ar,
+    return_index=False,
+    return_inverse=False,
+    return_counts=False,
+    axis=None,
+    *,
+    equal_nan=None,  # new in numpy 1.24
 ):
     unit = ar.unit
     n_index = sum(bool(i) for i in (return_index, return_inverse, return_counts))
     if n_index:
         unit = [unit] + n_index * [None]
 
-    return (ar.value, return_index, return_inverse, return_counts, axis), {}, unit, None
+    if NUMPY_LT_1_24:
+        if equal_nan is not None:
+            raise TypeError("unique() got an unexpected keyword argument 'equal_nan'")
```

Comment:
quoting a comment from @mhvk that I think was intended to land here (https://github.com/astropy/astropy/issues/15703#issuecomment-1848411973)
> In hindsight, I'm not sure we needed to introduce the checks in the wrapper, since for older numpy presumably the function call itself would raise a `TypeError`...

Reply:
So let me explain why I think this complicated logic is needed:
- if I strictly replicate the current (as of numpy 1.26) signature, I'll need to also replicate the default value `equal_nan=True`, but then I could not distinguish the case where the user left the value unspecified (which is valid in any version) from the case where the user explicitly passed `equal_nan=True` as a keyword argument (which is only valid in numpy 1.24+). So instead I use `None` as a sentinel value (I could even use a custom one to make extra sure that it will never be passed explicitly).
- in order to actually delegate exception raising to `numpy`, I'd need to accept arbitrary arguments here and just pass them on, which is what I did initially with `**kwargs`, but that wasn't satisfactory.

Does this second point answers your comment, @mhvk ?

Reply:
(I think we converged on this point in another thread but I'll let @mhvk a chance to add a comment or hit the resolve button)

Reply:
Yes, `**kwargs` after all!